src/main/scala/view/scalaFX/components/tree/GenealogicalTreeView.scala

```diff
@@ -0,0 +1,119 @@
+package view.scalaFX.components.tree
+
+import engine.SimulationConstants.MAX_GENEALOGICAL_TREE_GENERATIONS
+import model.Tree.generateTree
+import model.{BinaryTree, Bunny, Node}
+import scalafx.geometry.Pos
+import scalafx.scene.image.ImageView
+import scalafx.scene.layout._
+import scalafx.scene.text.Text
+import view.scalaFX.components.tree.GenealogicalTreeViewConstants.{BUNNY_PLUS_PROPORTION, BUNNY_REGION_PROPORTION, STANDARD_BUNNY_SIZE}
+
+/** Proportion constants to resize the view depending on the bunny size*/
+object GenealogicalTreeViewConstants {
+  val STANDARD_BUNNY_SIZE = 60
+  val BUNNY_INFO_PROPORTION: Int = 5
+  val BUNNY_REGION_PROPORTION: Int = 20
+  val BUNNY_FONT_PROPORTION: Int = 8
+  val BUNNY_PLUS_PROPORTION: Int = 3
+}
+
+trait GenealogicalTreeView{
+  /** Reference to the model bunny entity */
+  val bunny: Bunny
+
+  /** Reference to the model tree entity created from the bunny */
+  val tree: BinaryTree[Bunny]
+
+  /** The pane with the view of the tree */
+  val treePane: Pane
+}
+
+object GenealogicalTreeView {
+  /** The size required for the bunny icons*/
+  var bunnyIconSize: Int = STANDARD_BUNNY_SIZE
+
+  def apply(bunny: Bunny): GenealogicalTreeView = TreeViewImpl(bunny, generateTree(MAX_GENEALOGICAL_TREE_GENERATIONS, bunny))
+
+  def apply(bunny: Bunny, bunnySize: Int): GenealogicalTreeView = {
+    this.bunnyIconSize = bunnySize
+    this(bunny)
+  }
+
+  private case class TreeViewImpl(override val bunny: Bunny, override val tree: BinaryTree[Bunny]) extends GenealogicalTreeView{
+    var rows: Seq[HBox] = Seq()
+    var row: (HBox, Seq[Option[BinaryTree[Bunny]]]) = (new HBox(), Seq(Option(tree)))
+    for (_ <- 1 to tree.generations){
+      row = createRow(row._2)
+      rows ++= Seq(row._1)
```

Comment:
In alternativa a crearti ogni volta una Seq di un solo elemento potresti fare:  `rows = rows :+ row._1`

Reply:
Ecco qual era il simbolo per fare questa cosa! Grazie